src/pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -135,7 +135,7 @@ def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: O
             # validation, then we run after validation instead of on train epoch end
             self._check_on_train_epoch_end = trainer.val_check_interval == 1.0 and trainer.check_val_every_n_epoch == 1
 
-    def _validate_condition_metric(self, logs: Dict[str, float]) -> bool:
+    def _validate_condition_metric(self, logs: Dict[str, Tensor]) -> bool:
```

Comment:
`logs` should be the return type of `logger_connector.callback_metrics`

Reply:
I considered doing this. But didn't want to in this PR

The `callback_metrics` annotation is internal. I fear this creates circular imports and perhaps we should make it public and move it to `types.py`

I think this should be done in a follow-up